### PR TITLE
Add maven plugin for proto lock

### PR DESCRIPTION
**Pull request by timmylicheng**
#12596 
Introduce https://github.com/salesforce/proto-backwards-compat-maven-plugin to help prevent proto changes which can break backwards compatibility.
This plugin can fail the build when some field in a proto file is removed. Such as:

diff --git a/core/transport/src/main/proto/grpc/table/layout/hive/hive.proto b/core/transport/src/main/proto/grpc/table/layout/hive/hive.proto
index 538fbe02a0..829ed6f023 100644
--- a/core/transport/src/main/proto/grpc/table/layout/hive/hive.proto
+++ b/core/transport/src/main/proto/grpc/table/layout/hive/hive.proto
@@ -28,7 +28,6 @@ message SortingColumn {
ASCENDING = 0;
DESCENDING = 1;
}

required SortingOrder order = 2;
}
The plugin will trigger a build failure for notification:
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 27.064 s
[INFO] Finished at: 2020-12-22T21:45:30+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.salesforce.servicelibs:proto-backwards-compatibility:1.0.5:backwards-compatibility-check (default) on project alluxio-core-transport: Backwards compatibility check failed! -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR] mvn -rf :alluxio-core-transport

### CODE REVIEW

File: core/transport/pom.xml [hunk 1]
```diff
@@ -133,6 +134,23 @@
         </executions>
       </plugin>
 
+      <!-- Proto lock for preventing backwards compatibility issues -->
```
Comment (apc999) [inline core/transport/pom.xml:137]: please add definition of this artifact dependency  to `root/pom.xml` under 
`<dependencyManagement>`
https://github.com/Alluxio/alluxio/blob/0d149a8d397ac8c1b0092d390c85956706399ff2/pom.xml#L199
like
```
      <dependency>
        <groupId>com.salesforce.servicelibs</groupId>
        <artifactId>proto-backwards-compatibility</artifactId>
        <version>1.0.5</version>
      </dependency>
```

Then in this `core/transport/pom.xml`, you only need to refer to the artifact without specifying the version
```
      <plugin>
        <groupId>com.salesforce.servicelibs</groupId>
        <artifactId>proto-backwards-compatibility</artifactId>
        <configuration>
          <protoSourceRoot>${basedir}/target/classes/</protoSourceRoot>
        </configuration>
        <executions>
          <execution>
            <goals>
              <goal>backwards-compatibility-check</goal>
            </goals>
          </execution>
        </executions>
      </plugin>
```

Comment (timmylicheng) [reply core/transport/pom.xml:136]: Thanks for the tip. I've updated the PR.

**Comment by timmylicheng** [comment]
@apc999 Hey Bin, looks like there is something failing the tests.
2020-12-23T22:01:27.8430330Z java.lang.NullPointerException
2020-12-23T22:01:27.8432008Z 	at alluxio.master.metastore.caching.CachingInodeStore$EdgeCache.lambda$verifyIndices$8(CachingInodeStore.java:530)

Sorry I'm still not familiar with alluxio tests. Any clue on why pom change would trigger this?

**Comment by maobaolong** [comment]
@timmylicheng Maybe it is a flaky test, so i suggest you fix this test or try to close and open this PR again to trigger the CI check again.

**Review by apc999 — APPROVED** [review]

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**